### DISCUSSION TITLE
Composing Files for Debian Bookworm via Balena Cloud

### DISCUSSION THREAD

**Discussion by CRSBenson** [Screenly/Anthias#1884] [Q&A]
Hi All,

To start, I'm new to the Balena Cloud, Docker components, and posting in GitHub, but I'm picking up on them fairly quickly. I don't know if this post goes here or in pull requests. Anthias is similar to what I have developed for an RTSP viewer on a monitor that runs 12 hours daily, CEC, error reporting, etc., but without the power of Docker, things began to interfere. After deciding to ditch the custom Python script I was writing, I found Anthias, and I would love to help the project and make it better for all. 

I've made some adjustments to the docker files to attempt to upload to my fleet via Balena cloud, which was fine when sticking with the lib files based on Buster. Now that I've been migrating the files to pull from the /lib/bookworm section for ARMv8 devices, I'm getting stuck when using ./bin/upload_to_balena.sh --board pi4 --fleet [Fleet Name] --dev

Upon running the command, everything seems to compile when accessing the docker library files. However, when the containers are building, apt-get is having some issues. I'll post the log files below and attach them in a file as well for easier review. Is there something different with Bookworm that would prevent the use of "apt-get" or the combo of "apt-get with /bin/sh -c" as seen in the log for multiple containers? Is this because I don't have the Anthias repo mounted as a docker volume as stated in the readme or because I'm running this on WSL2 (even though I have x86 architecture tagged to run as a Pi 4 board when detected?

 '/bin/sh -c apt-get update &&     apt-get -y install --no-install-recommends....... ' returned a non-zero code: 100

I'm really stumped on this one and would appreciate any assistance with getting the contains to build correctly then I'll be able to continue testing something that we can hopefully use in the master for dev templates when other users are deploying their custom builds to Balena. 

Note: the .git file was removed from the Anthias root folder due to its size. 

[Anthias.zip](https://github.com/Screenly/Anthias/files/15029525/Anthias.zip)
[Deploy to Balena on Bookworm logs.txt](https://github.com/Screenly/Anthias/files/15029550/Deploy.to.Balena.on.Bookworm.logs.txt)

Running in dev mode...
++ git rev-parse --abbrev-ref HEAD
+ export GIT_BRANCH=master
+ GIT_BRANCH=master
++ git rev-parse --short HEAD
+ export GIT_SHORT_HASH=503498e
+ GIT_SHORT_HASH=503498e
++ git rev-parse HEAD
+ export GIT_HASH=503498e92ff06a98d6b7e89491d87c29180e1585
+ GIT_HASH=503498e92ff06a98d6b7e89491d87c29180e1585
+ export BASE_IMAGE_TAG=bookworm
+ BASE_IMAGE_TAG=bookworm
+ export DEBIAN_VERSION=bookworm
+ DEBIAN_VERSION=bookworm
+ SERVICES=('server' 'celery' 'redis' 'websocket' 'nginx' 'viewer' 'wifi-connect' 'test')
+ declare -a SERVICES
+ DOCKER_BUILD_ARGS=("buildx" "build" "--load")
+ echo 'Make sure you ran `docker buildx create --use` before the command'
Make sure you ran `docker buildx create --use` before the command
+ '[' -n '' ']'
+ '[' '!' -f /proc/device-tree/model ']'
+ '[' -z x ']'
+ grep -qF 'Raspberry Pi 4' /proc/device-tree/model
grep: /proc/device-tree/model: No such file or directory
+ '[' pi4 == pi4 ']'
+ export BASE_IMAGE=balenalib/raspberrypi4-64-debian
+ BASE_IMAGE=balenalib/raspberrypi4-64-debian
+ export BOARD=pi4
+ BOARD=pi4
+ export TARGET_PLATFORM=linux/arm64
+ TARGET_PLATFORM=linux/arm64
+ '[' master = master ']'
+ export DOCKER_TAG=latest-pi4
+ DOCKER_TAG=latest-pi4
+ for container in ${SERVICES[@]}
+ echo 'Building server...'
Building server...
++ echo server
++ tr '[:lower:]' '[:upper:]'
++ tr - _
+ uppercase_container=SERVER
+ skip_variable=SKIP_SERVER
+ '[' -n '' ']'
+ '[' server == viewer ']'
+ '[' server == test ']'
+ '[' server == wifi-connect ']'
+ [[ ! server =~ ^(redis|nginx|viewer)$ ]]
+ cat docker/Dockerfile.base.tmpl
+ envsubst
+ cat docker/Dockerfile.server.tmpl
+ envsubst
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ sed -i 's/RUN --mount.\+ /RUN /g' docker/Dockerfile.server
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ echo 'Variable DOCKERFILES_ONLY is set. Skipping build for server...'
Variable DOCKERFILES_ONLY is set. Skipping build for server...
+ continue
+ for container in ${SERVICES[@]}
+ echo 'Building celery...'
Building celery...
++ echo celery
++ tr '[:lower:]' '[:upper:]'
++ tr - _
+ uppercase_container=CELERY
+ skip_variable=SKIP_CELERY
+ '[' -n '' ']'
+ '[' celery == viewer ']'
+ '[' celery == test ']'
+ '[' celery == wifi-connect ']'
+ [[ ! celery =~ ^(redis|nginx|viewer)$ ]]
+ cat docker/Dockerfile.base.tmpl
+ envsubst
+ cat docker/Dockerfile.celery.tmpl
+ envsubst
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ sed -i 's/RUN --mount.\+ /RUN /g' docker/Dockerfile.celery
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ echo 'Variable DOCKERFILES_ONLY is set. Skipping build for celery...'
Variable DOCKERFILES_ONLY is set. Skipping build for celery...
+ continue
+ for container in ${SERVICES[@]}
+ echo 'Building redis...'
Building redis...
++ echo redis
++ tr '[:lower:]' '[:upper:]'
++ tr - _
+ uppercase_container=REDIS
+ skip_variable=SKIP_REDIS
+ '[' -n '' ']'
+ '[' redis == viewer ']'
+ '[' redis == test ']'
+ '[' redis == wifi-connect ']'
+ [[ ! redis =~ ^(redis|nginx|viewer)$ ]]
+ cat docker/Dockerfile.redis.tmpl
+ envsubst
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ sed -i 's/RUN --mount.\+ /RUN /g' docker/Dockerfile.redis
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ echo 'Variable DOCKERFILES_ONLY is set. Skipping build for redis...'
Variable DOCKERFILES_ONLY is set. Skipping build for redis...
+ continue
+ for container in ${SERVICES[@]}
+ echo 'Building websocket...'
Building websocket...
++ echo websocket
++ tr '[:lower:]' '[:upper:]'
++ tr - _
+ uppercase_container=WEBSOCKET
+ skip_variable=SKIP_WEBSOCKET
+ '[' -n '' ']'
+ '[' websocket == viewer ']'
+ '[' websocket == test ']'
+ '[' websocket == wifi-connect ']'
+ [[ ! websocket =~ ^(redis|nginx|viewer)$ ]]
+ cat docker/Dockerfile.base.tmpl
+ envsubst
+ cat docker/Dockerfile.websocket.tmpl
+ envsubst
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ sed -i 's/RUN --mount.\+ /RUN /g' docker/Dockerfile.websocket
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ echo 'Variable DOCKERFILES_ONLY is set. Skipping build for websocket...'
Variable DOCKERFILES_ONLY is set. Skipping build for websocket...
+ continue
+ for container in ${SERVICES[@]}
+ echo 'Building nginx...'
Building nginx...
++ echo nginx
++ tr '[:lower:]' '[:upper:]'
++ tr - _
+ uppercase_container=NGINX
+ skip_variable=SKIP_NGINX
+ '[' -n '' ']'
+ '[' nginx == viewer ']'
+ '[' nginx == test ']'
+ '[' nginx == wifi-connect ']'
+ [[ ! nginx =~ ^(redis|nginx|viewer)$ ]]
+ cat docker/Dockerfile.nginx.tmpl
+ envsubst
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ sed -i 's/RUN --mount.\+ /RUN /g' docker/Dockerfile.nginx
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ echo 'Variable DOCKERFILES_ONLY is set. Skipping build for nginx...'
Variable DOCKERFILES_ONLY is set. Skipping build for nginx...
+ continue
+ for container in ${SERVICES[@]}
+ echo 'Building viewer...'
Building viewer...
++ echo viewer
++ tr '[:lower:]' '[:upper:]'
++ tr - _
+ uppercase_container=VIEWER
+ skip_variable=SKIP_VIEWER
+ '[' -n '' ']'
+ '[' viewer == viewer ']'
+ export QT_VERSION=5.15.2
+ QT_VERSION=5.15.2
+ export WEBVIEW_GIT_HASH=f5ef562982dcb6274c9716b9e375cc5ac0faba84
+ WEBVIEW_GIT_HASH=f5ef562982dcb6274c9716b9e375cc5ac0faba84
+ export WEBVIEW_BASE_URL=https://github.com/Screenly/Anthias/releases/download/WebView-v0.2.2
+ WEBVIEW_BASE_URL=https://github.com/Screenly/Anthias/releases/download/WebView-v0.2.2
+ [[ ! viewer =~ ^(redis|nginx|viewer)$ ]]
+ cat docker/Dockerfile.viewer.tmpl
+ envsubst
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ sed -i 's/RUN --mount.\+ /RUN /g' docker/Dockerfile.viewer
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ echo 'Variable DOCKERFILES_ONLY is set. Skipping build for viewer...'
Variable DOCKERFILES_ONLY is set. Skipping build for viewer...
+ continue
+ for container in ${SERVICES[@]}
+ echo 'Building wifi-connect...'
Building wifi-connect...
++ echo wifi-connect
++ tr '[:lower:]' '[:upper:]'
++ tr - _
+ uppercase_container=WIFI_CONNECT
+ skip_variable=SKIP_WIFI_CONNECT
+ '[' -n '' ']'
+ '[' wifi-connect == viewer ']'
+ '[' wifi-connect == test ']'
+ '[' wifi-connect == wifi-connect ']'
+ '[' pi4 == x86 ']'
+ '[' linux/arm64 = linux/arm/v6 ']'
+ architecture=armv7hf
+ wc_download_url=https://api.github.com/repos/balena-os/wifi-connect/releases/93025295
+ jq_filter='.assets[] | select (.name|test("linux-armv7hf")) | .browser_download_url'
++ curl -sL https://api.github.com/repos/balena-os/wifi-connect/releases/93025295
++ jq -r '.assets[] | select (.name|test("linux-armv7hf")) | .browser_download_url'
+ archive_url=https://github.com/balena-os/wifi-connect/releases/download/v4.11.1/wifi-connect-v4.11.1-linux-armv7hf.zip
+ export ARCHIVE_URL=https://github.com/balena-os/wifi-connect/releases/download/v4.11.1/wifi-connect-v4.11.1-linux-armv7hf.zip
+ ARCHIVE_URL=https://github.com/balena-os/wifi-connect/releases/download/v4.11.1/wifi-connect-v4.11.1-linux-armv7hf.zip
+ [[ ! wifi-connect =~ ^(redis|nginx|viewer)$ ]]
+ cat docker/Dockerfile.base.tmpl
+ envsubst
+ cat docker/Dockerfile.wifi-connect.tmpl
+ envsubst
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ sed -i 's/RUN --mount.\+ /RUN /g' docker/Dockerfile.wifi-connect
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ echo 'Variable DOCKERFILES_ONLY is set. Skipping build for wifi-connect...'
Variable DOCKERFILES_ONLY is set. Skipping build for wifi-connect...
+ continue
+ for container in ${SERVICES[@]}
+ echo 'Building test...'
Building test...
++ echo test
++ tr '[:lower:]' '[:upper:]'
++ tr - _
+ uppercase_container=TEST
+ skip_variable=SKIP_TEST
+ '[' -n '' ']'
+ '[' test == viewer ']'
+ '[' test == test ']'
+ export CHROME_DL_URL=https://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_107.0.5304.121-1_amd64.deb
+ CHROME_DL_URL=https://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_107.0.5304.121-1_amd64.deb
+ export CHROMEDRIVER_DL_URL=https://chromedriver.storage.googleapis.com/107.0.5304.62/chromedriver_linux64.zip
+ CHROMEDRIVER_DL_URL=https://chromedriver.storage.googleapis.com/107.0.5304.62/chromedriver_linux64.zip
+ [[ ! test =~ ^(redis|nginx|viewer)$ ]]
+ cat docker/Dockerfile.base.tmpl
+ envsubst
+ cat docker/Dockerfile.test.tmpl
+ envsubst
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ sed -i 's/RUN --mount.\+ /RUN /g' docker/Dockerfile.test
+ [[ -n 1 ]]
+ [[ 1 -ne 0 ]]
+ echo 'Variable DOCKERFILES_ONLY is set. Skipping build for test...'
Variable DOCKERFILES_ONLY is set. Skipping build for test...
+ continue
[Info]                  Starting build for gps-secured-entryways, user chandlers
[Info]                  Dashboard link: https://dashboard.balena-cloud.com/apps/2122177/devices
[Info]                  Building on 4b2d7d0
[Info]                  Pulling previous images for caching purposes...
[Success]               Successfully pulled cache images
[anthias-nginx]         Step 1/8 : FROM balenalib/raspberrypi4-64-debian:bookworm
[redis]                 Step 1/8 : FROM balenalib/raspberrypi4-64-debian:bookworm
[anthias-server]        Step 1/12 : FROM balenalib/raspberrypi4-64-debian:bookworm
[anthias-wifi-connect]  Step 1/16 : FROM balenalib/raspberrypi4-64-debian:bookworm
[anthias-nginx]          ---> bc3fc5372876
[anthias-nginx]         Step 2/8 : RUN     apt-get update &&     apt-get -y install --no-install-recommends     nginx
[anthias-nginx]          ---> Running in ad4d11284854
[anthias-celery]        Step 1/12 : FROM balenalib/raspberrypi4-64-debian:bookworm
[redis]                  ---> bc3fc5372876
[redis]                 Step 2/8 : RUN     apt-get update &&     apt-get -y install --no-install-recommends     redis-server
[redis]                  ---> Running in b1cf3386cfc4
[anthias-server]         ---> bc3fc5372876
[anthias-server]        Step 2/12 : RUN     apt-get update &&     apt-get -y install --no-install-recommends         build-essential         cec-utils         curl         ffmpeg         git         git-core         ifupdown         libcec-dev          libffi-dev         libraspberrypi0         libraspberrypi0         libssl-dev         libzmq3-dev         libzmq5-dev         libzmq5         lsb-release         mplayer         net-tools         omxplayer         procps         psmisc         python3-dev         python3-gi         python3-pil         python3-pip         python3-setuptools         python3-simplejson         python3-full         sqlite3
[anthias-server]         ---> Running in fe70bacf1099
[anthias-wifi-connect]   ---> bc3fc5372876
[anthias-wifi-connect]  Step 2/16 : RUN     apt-get update &&     apt-get -y install --no-install-recommends         build-essential         cec-utils         curl         ffmpeg         git         git-core         ifupdown         libcec-dev          libffi-dev         libraspberrypi0         libraspberrypi0         libssl-dev         libzmq3-dev         libzmq5-dev         libzmq5         lsb-release         mplayer         net-tools         omxplayer         procps         psmisc         python3-dev         python3-gi         python3-pil         python3-pip         python3-setuptools         python3-simplejson         python3-full         sqlite3
[anthias-wifi-connect]   ---> Running in 20860d71cf1b
[anthias-celery]         ---> bc3fc5372876
[anthias-celery]        Step 2/12 : RUN     apt-get update &&     apt-get -y install --no-install-recommends         build-essential         cec-utils         curl         ffmpeg         git         git-core         ifupdown         libcec-dev          libffi-dev         libraspberrypi0         libraspberrypi0         libssl-dev         libzmq3-dev         libzmq5-dev         libzmq5         lsb-release         mplayer         net-tools         omxplayer         procps         psmisc         python3-dev         python3-gi         python3-pil         python3-pip         python3-setuptools         python3-simplejson         python3-full         sqlite3
[anthias-celery]         ---> Running in 9710e92135eb
[redis]                 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
[anthias-nginx]         Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
[redis]                 Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
[redis]                 Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
[anthias-nginx]         Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
[anthias-nginx]         Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
[anthias-server]        Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
[anthias-server]        Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
[anthias-server]        Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
[redis]                 Get:4 http://deb.debian.org/debian bookworm/main arm64 Packages [8685 kB]
[anthias-nginx]         Get:4 http://deb.debian.org/debian bookworm/main arm64 Packages [8685 kB]
[anthias-server]        Get:4 http://deb.debian.org/debian bookworm/main arm64 Packages [8685 kB]
[redis]                 Get:5 http://deb.debian.org/debian bookworm-updates/main arm64 Packages [12.5 kB]
[redis]                 Get:6 http://deb.debian.org/debian-security bookworm-security/main arm64 Packages [151 kB]
[anthias-server]        Get:5 http://deb.debian.org/debian bookworm-updates/main arm64 Packages [12.5 kB]
[anthias-nginx]         Get:5 http://deb.debian.org/debian bookworm-updates/main arm64 Packages [12.5 kB]
[anthias-nginx]         Get:6 http://deb.debian.org/debian-security bookworm-security/main arm64 Packages [151 kB]
[anthias-wifi-connect]  Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
[anthias-celery]        Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
[anthias-wifi-connect]  Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
[anthias-wifi-connect]  Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
[anthias-celery]        Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
[anthias-celery]        Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
[anthias-server]        Get:6 http://deb.debian.org/debian-security bookworm-security/main arm64 Packages [151 kB]
[anthias-wifi-connect]  Get:4 http://deb.debian.org/debian bookworm/main arm64 Packages [8685 kB]
[anthias-celery]        Get:4 http://deb.debian.org/debian bookworm/main arm64 Packages [8685 kB]
[anthias-wifi-connect]  Get:5 http://deb.debian.org/debian bookworm-updates/main arm64 Packages [12.5 kB]
[anthias-celery]        Get:5 http://deb.debian.org/debian bookworm-updates/main arm64 Packages [12.5 kB]
[anthias-wifi-connect]  Get:6 http://deb.debian.org/debian-security bookworm-security/main arm64 Packages [151 kB]
[anthias-celery]        Get:6 http://deb.debian.org/debian-security bookworm-security/main arm64 Packages [151 kB]
[redis]                 Fetched 9103 kB in 1s (8026 kB/s)
[redis]                 Reading package lists...
[anthias-server]        Fetched 9103 kB in 1s (7922 kB/s)
[anthias-server]        Reading package lists...
[anthias-nginx]         Fetched 9103 kB in 1s (6927 kB/s)
[anthias-nginx]         Reading package lists...
[anthias-wifi-connect]  Fetched 9103 kB in 1s (7631 kB/s)
[anthias-wifi-connect]  Reading package lists...
[anthias-celery]        Fetched 9103 kB in 1s (7505 kB/s)
[anthias-celery]        Reading package lists...
[redis]                 Reading package lists...
[anthias-server]        Reading package lists...
[anthias-nginx]         Reading package lists...
[anthias-wifi-connect]  Reading package lists...
[anthias-celery]        Reading package lists...
[anthias-websocket]     Step 1/12 : FROM balenalib/raspberrypi4-64-debian:bookworm
[redis]                 Building dependency tree...
[anthias-nginx]         Building dependency tree...
[anthias-server]        Building dependency tree...
[redis]                 Reading state information...
[anthias-nginx]         Reading state information...
[anthias-server]        Reading state information...
[anthias-server]        Package omxplayer is not available, but is referred to by another package.
[anthias-server]        This may mean that the package is missing, has been obsoleted, or
[anthias-server]        is only available from another source
[anthias-server]        E: Unable to locate package libraspberrypi0
[anthias-server]        E: Unable to locate package libraspberrypi0
[anthias-server]        E: Package 'omxplayer' has no installation candidate
[anthias-server]        
[redis]                 The following additional packages will be installed:
[redis]                   libatomic1 libjemalloc2 liblzf1 redis-tools
[redis]                 Suggested packages:
[redis]                   ruby-redis
[redis]                 The following NEW packages will be installed:
[redis]                   libatomic1 libjemalloc2 liblzf1 redis-server redis-tools
[anthias-wifi-connect]  Building dependency tree...
[anthias-celery]        Building dependency tree...
[anthias-nginx]         The following additional packages will be installed:
[anthias-nginx]           nginx-common
[anthias-nginx]         Suggested packages:
[anthias-nginx]           fcgiwrap nginx-doc ssl-cert
[anthias-nginx]         The following NEW packages will be installed:
[anthias-nginx]           nginx nginx-common
[redis]                 0 upgraded, 5 newly installed, 0 to remove and 8 not upgraded.
[redis]                 Need to get 1205 kB of archives.
[redis]                 After this operation, 7074 kB of additional disk space will be used.
[redis]                 Get:1 http://deb.debian.org/debian bookworm/main arm64 libatomic1 arm64 12.2.0-14 [9536 B]
[redis]                 Get:2 http://deb.debian.org/debian bookworm/main arm64 libjemalloc2 arm64 5.3.0-1 [217 kB]
[redis]                 Get:3 http://deb.debian.org/debian bookworm/main arm64 liblzf1 arm64 3.6-3 [9860 B]
[redis]                 Get:4 http://deb.debian.org/debian bookworm/main arm64 redis-tools arm64 5:7.0.15-1~deb12u1 [897 kB]
[redis]                 Get:5 http://deb.debian.org/debian bookworm/main arm64 redis-server arm64 5:7.0.15-1~deb12u1 [72.4 kB]
[anthias-nginx]         0 upgraded, 2 newly installed, 0 to remove and 8 not upgraded.
[anthias-nginx]         Need to get 606 kB of archives.
[anthias-nginx]         After this operation, 1708 kB of additional disk space will be used.
[anthias-nginx]         Get:1 http://deb.debian.org/debian bookworm/main arm64 nginx-common all 1.22.1-9 [112 kB]
[anthias-nginx]         Get:2 http://deb.debian.org/debian bookworm/main arm64 nginx arm64 1.22.1-9 [494 kB]
[anthias-wifi-connect]  Reading state information...
[anthias-celery]        Reading state information...
[anthias-celery]        Package omxplayer is not available, but is referred to by another package.
[anthias-celery]        This may mean that the package is missing, has been obsoleted, or
[anthias-celery]        is only available from another source
[anthias-wifi-connect]  Package omxplayer is not available, but is referred to by another package.
[anthias-wifi-connect]  This may mean that the package is missing, has been obsoleted, or
[anthias-wifi-connect]  is only available from another source
[anthias-celery]        E: Unable to locate package libraspberrypi0
[anthias-celery]        E: Unable to locate package libraspberrypi0
[anthias-celery]        E: Package 'omxplayer' has no installation candidate
[anthias-celery]        
[anthias-wifi-connect]  E: Unable to locate package libraspberrypi0
[anthias-wifi-connect]  E: Unable to locate package libraspberrypi0
[anthias-wifi-connect]  E: Package 'omxplayer' has no installation candidate
[anthias-wifi-connect]  
[redis]                 debconf: delaying package configuration, since apt-utils is not installed
[redis]                 
[redis]                 Fetched 1205 kB in 0s (9743 kB/s)
[anthias-nginx]         debconf: delaying package configuration, since apt-utils is not installed
[anthias-nginx]         
[redis]                 Selecting previously unselected package libatomic1:arm64.
(Reading database ... 85%Reading database ... 
[redis]                 (Reading database ... 90%
[redis]                 (Reading database ... 95%
(Reading database ... 10030 files and directories currently installed.)
[redis]                 Preparing to unpack .../libatomic1_12.2.0-14_arm64.deb ...
[redis]                 Unpacking libatomic1:arm64 (12.2.0-14) ...
[redis]                 Selecting previously unselected package libjemalloc2:arm64.
[redis]                 Preparing to unpack .../libjemalloc2_5.3.0-1_arm64.deb ...
[redis]                 Unpacking libjemalloc2:arm64 (5.3.0-1) ...
[anthias-nginx]         Fetched 606 kB in 0s (6687 kB/s)
[anthias-nginx]         Selecting previously unselected package nginx-common.
[anthias-nginx]         (Reading database ... 
(Reading database ... 70%Reading database ... 5%
[anthias-nginx]         (Reading database ... 75%
[anthias-nginx]         (Reading database ... 80%
[anthias-nginx]         (Reading database ... 85%
[anthias-nginx]         (Reading database ... 90%
[anthias-nginx]         (Reading database ... 95%
(Reading database ... 10030 files and directories currently installed.)
[redis]                 Selecting previously unselected package liblzf1:arm64.
[redis]                 Preparing to unpack .../liblzf1_3.6-3_arm64.deb ...
[redis]                 Unpacking liblzf1:arm64 (3.6-3) ...
[anthias-nginx]         Preparing to unpack .../nginx-common_1.22.1-9_all.deb ...
[anthias-nginx]         Unpacking nginx-common (1.22.1-9) ...
[redis]                 Selecting previously unselected package redis-tools.
[redis]                 Preparing to unpack .../redis-tools_5%3a7.0.15-1~deb12u1_arm64.deb ...
[redis]                 Unpacking redis-tools (5:7.0.15-1~deb12u1) ...
[anthias-nginx]         Selecting previously unselected package nginx.
[anthias-nginx]         Preparing to unpack .../nginx_1.22.1-9_arm64.deb ...
[anthias-nginx]         Unpacking nginx (1.22.1-9) ...
[anthias-nginx]         Setting up nginx-common (1.22.1-9) ...
[redis]                 Selecting previously unselected package redis-server.
[redis]                 Preparing to unpack .../redis-server_5%3a7.0.15-1~deb12u1_arm64.deb ...
[redis]                 Unpacking redis-server (5:7.0.15-1~deb12u1) ...
[redis]                 Setting up libjemalloc2:arm64 (5.3.0-1) ...
[redis]                 Setting up liblzf1:arm64 (3.6-3) ...
[redis]                 Setting up libatomic1:arm64 (12.2.0-14) ...
[redis]                 Setting up redis-tools (5:7.0.15-1~deb12u1) ...
[anthias-websocket]      ---> bc3fc5372876
[anthias-websocket]     Step 2/12 : RUN     apt-get update &&     apt-get -y install --no-install-recommends         build-essential         cec-utils         curl         ffmpeg         git         git-core         ifupdown         libcec-dev          libffi-dev         libraspberrypi0         libraspberrypi0         libssl-dev         libzmq3-dev         libzmq5-dev         libzmq5         lsb-release         mplayer         net-tools         omxplayer         procps         psmisc         python3-dev         python3-gi         python3-pil         python3-pip         python3-setuptools         python3-simplejson         python3-full         sqlite3
[anthias-websocket]      ---> Running in 2d601a58676c
[anthias-server]        Removing intermediate container fe70bacf1099
[anthias-server]        The command '/bin/sh -c apt-get update &&     apt-get -y install --no-install-recommends         build-essential         cec-utils         curl         ffmpeg         git         git-core         ifupdown         libcec-dev          libffi-dev         libraspberrypi0         libraspberrypi0         libssl-dev         libzmq3-dev         libzmq5-dev         libzmq5         lsb-release         mplayer         net-tools         omxplayer         procps         psmisc         python3-dev         python3-gi         python3-pil         python3-pip         python3-setuptools         python3-simplejson         python3-full         sqlite3' returned a non-zero code: 100
[redis]                 Setting up redis-server (5:7.0.15-1~deb12u1) ...
[anthias-nginx]         Setting up nginx (1.22.1-9) ...
[redis]                 invoke-rc.d: could not determine current runlevel
[redis]                 invoke-rc.d: policy-rc.d denied execution of start.
[anthias-nginx]         invoke-rc.d: could not determine current runlevel
[anthias-nginx]         invoke-rc.d: policy-rc.d denied execution of start.
[redis]                 Processing triggers for libc-bin (2.36-9+deb12u4) ...
[anthias-celery]        Removing intermediate container 9710e92135eb
[anthias-celery]        The command '/bin/sh -c apt-get update &&     apt-get -y install --no-install-recommends         build-essential         cec-utils         curl         ffmpeg         git         git-core         ifupdown         libcec-dev          libffi-dev         libraspberrypi0         libraspberrypi0         libssl-dev         libzmq3-dev         libzmq5-dev         libzmq5         lsb-release         mplayer         net-tools         omxplayer         procps         psmisc         python3-dev         python3-gi         python3-pil         python3-pip         python3-setuptools         python3-simplejson         python3-full         sqlite3' returned a non-zero code: 100
[anthias-wifi-connect]  Removing intermediate container 20860d71cf1b
[anthias-wifi-connect]  The command '/bin/sh -c apt-get update &&     apt-get -y install --no-install-recommends         build-essential         cec-utils         curl         ffmpeg         git         git-core         ifupdown         libcec-dev          libffi-dev         libraspberrypi0         libraspberrypi0         libssl-dev         libzmq3-dev         libzmq5-dev         libzmq5         lsb-release         mplayer         net-tools         omxplayer         procps         psmisc         python3-dev         python3-gi         python3-pil         python3-pip         python3-setuptools         python3-simplejson         python3-full         sqlite3' returned a non-zero code: 100
^C
Received SIGINT, cleaning up. Please wait.
Setting 'cancelled' release status for release ID 2975155 ...
Error: aborted
    at connResetException (node:internal/errors:787:14)
    at TLSSocket.socketCloseListener (node:_http_client:455:19)
    at TLSSocket.emit (node:events:526:35)
    at TLSSocket.emit (node:domain:488:12)
    at node:net:337:12
    at TCP.done (node:_tls_wrap:657:7)

**Comment by nicomiguelino** [comment]
There are some dependencies/packages in Buster that are not present in Bullseye (and vice versa).
Since they're major releases, there are breaking changes. It can be overwhelming to fix all the Dockerfiles (for each of the services) at once.
* It would be better to create a pull request for this. That way, other contributors can see the changes.
* With that you can also see which unit tests and CI jobs are failing.
* You can get around with it step by step.
  * Focus on one Dockerfile at a time. For example, you can start with the server Dockerfile.
  * Comment the Debian packages that cannot be installed.
  * There might be some packages in Bookworm that replaces certain packages in Buster, so you might want to go through the dependencies line by line.
